### PR TITLE
Hash link converter and `/embed` and `/postembed` refactor

### PR DESCRIPTION
**Pull request by Fritz-02**
Added a HashLinkConverter class to handle converting URLs to an embed hash link.
Refactored `/embed` and `/postembed` commands to use this class. (#149)

### CODE REVIEW

File: assets/chat/js/chat.js [hunk 1]
```diff
@@ -2163,182 +2167,76 @@ class Chat {
   cmdEMBED(parts) {
```
Comment (11k) [inline assets/chat/js/chat.js:2167]: I think having separate error messages related to the number of arguments is unnecessary. Try to simplify all this such that you just print two things on invalid use of the command: the error from the convertor, and a usage hint.

**Review by 11k — CHANGES_REQUESTED** [review]

**Comment by Fritz-02** [comment]
Error outputs for the commands, to show what they look like. The info message about usage for `/postembed` might be too lengthy, so it could be split across two messages.

Screenshot from 2023-02-15 05-29-09
Screenshot from 2023-02-15 05-47-08

**Comment by 11k** [comment]
The usage hint for `/postembed` is lengthy because it relists the arguments for the alias, and lists the command without arguments as "alternative syntax" without acknowledging that `<link>` is optional. Something like
```
Usage: /postembed [<link>] [<message>] (Alias: /pe) (Valid links: Twitch streams, VODs, clips, Youtube, Rumble)
```
addresses this.

To take a step back, it's a relatively complex command whose behavior can't be conveyed in a single line. It might be better to explain in more detail what it does if the user uses it incorrectly, but that's a problem for another day.

**Review by 11k — CHANGES_REQUESTED** [review]

File: assets/chat/js/chat.js [hunk 2]
```diff
@@ -2163,181 +2167,42 @@ class Chat {
...
-      MessageBuilder.info(
-        'Valid links: Twitch Streams, Twitch VODs, Twitch Clips, Youtube Videos, Vimeo Videos, Rumble Videos.'
-      ).into(this);
-    } else if (parts.length > 1) {
-      MessageBuilder.error(
-        'More than one argument provided - /embed <link> OR /e <link>'
-      ).into(this);
+    try {
+      const hashLink = this.hashLinkConverter.convert(parts[0]);
+      location.href = `${noEmbedUrl}${hashLink}`;
```
[the remark this answers is not in the document]
Comment (Fritz-02) [reply assets/chat/js/chat.js:2172]: Nice, works for me and looks cleaner. Done 👍

**Review by 11k — APPROVED** [review]

**Pull request merged**